webapp/app/js/services/main.js

```diff
@@ -184,12 +184,24 @@ treeherder.factory('thNotify', [
             $log.debug("received message", message);
             var severity = severity || 'info';
             var sticky = sticky || false;
+            var maxNsNotifications = 5;
             thNotify.notifications.push({
                 message: message,
                 severity: severity,
                 sticky: sticky
             });
-            if(!sticky){
+
+            if (!sticky) {
+                if (thNotify.notifications.length > maxNsNotifications) {
+                    /* Replicate shift code to avoid pruning during rapid retriggers
+                     * otheriwse we could use $timeout(thNotify.shift) here */
+                    for(var i=0;i<thNotify.notifications.length; i++) {
```

Comment:
I wonder if reusing the ``shift`` code didn't work because of that ``return`` there.  If you just call ``shift`` it will return from shift, but not from this code.  which would mean it would still call the ``$timeout`` on line 205.  Maybe that was the problem?